File: src/main/java/dev/zprestige/fire/module/movement/speed/TickListener.java
package dev.zprestige.fire.module.movement.speed;

import dev.zprestige.fire.Main;
import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.TickEvent;

import java.util.HashMap;
import java.util.Map;

public class TickListener extends EventListener<TickEvent, Speed> {

    public TickListener(final Speed speed) {
        super(TickEvent.class, speed);
    }

    @Override
    public void invoke(final Object object) {
        module.previousDistance = Math.sqrt((mc.player.posX - mc.player.prevPosX) * (mc.player.posX - mc.player.prevPosX) + (mc.player.posZ - mc.player.prevPosZ) * (mc.player.posZ - mc.player.prevPosZ));
        if (module.useTimer.GetSwitch()) {
            Main.tickManager.setTimer(module.timerAmount.GetSlider());
        }
        final float health = mc.player.getHealth() + mc.player.getAbsorptionAmount();
        float damage = 0.0f;
        for (Map.Entry<Long, Float> entry : new HashMap<>(module.damageMap).entrySet()) {
            double val = entry.getValue() / (((System.currentTimeMillis() - entry.getKey())) / module.boostAmplifier.GetSlider());
            if (val < 0.1) {
                module.damageMap.remove(entry.getKey());
            }
            damage += val;
        }
        final float dmg = module.lastHealth - health;
        if (dmg > 0) {
            module.damageMap.put(System.currentTimeMillis(), dmg);
        }
        module.lastHealth = mc.player.getHealth() + mc.player.getAbsorptionAmount();
        if (module.strict.GetSwitch()) {
            module.strafeFactor.setValue(0.9f);
        } else {
            if (module.velocityBoost.GetSwitch()) {
                module.strafeFactor.setValue(Math.max(1.0f, damage));
            }
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/movement/speed/TickListener.kt
package dev.zprestige.fire.module.movement.speed

import dev.zprestige.fire.Main
import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.TickEvent
import kotlin.math.max
import kotlin.math.sqrt

class TickListener(speed: Speed) : EventListener<TickEvent, Speed>(TickEvent::class.java, speed) {
    
    override fun invoke(e: Any) {
        module.previousDistance = sqrt((mc.player.posX - mc.player.prevPosX) * (mc.player.posX - mc.player.prevPosX) + (mc.player.posZ - mc.player.prevPosZ) * (mc.player.posZ - mc.player.prevPosZ))
        if (module.useTimer.GetSwitch()) {
            Main.tickManager.setTimer(module.timerAmount.GetSlider())
        }
        val health = mc.player.health + mc.player.absorptionAmount
        var damage = 0.0f
        for (i in module.damageMap){
            val v: Float = i.value / ((System.currentTimeMillis() - i.key) / module.boostAmplifier.GetSlider())
            if (v < 0.1) {
                module.damageMap.remove(i.key)
            }
            damage += v
        }

        val dmg: Float = module.lastHealth.minus(health)
        if (dmg > 0.0f) {
            module.damageMap[System.currentTimeMillis()] = dmg
        }
        module.lastHealth = mc.player.health + mc.player.absorptionAmount
        if (module.strict.GetSwitch()) {
            module.strafeFactor.value = 0.9f
        } else {
            if (module.velocityBoost.GetSwitch()) {
                module.strafeFactor.value = max(1.0f, damage)
            }
        }
    }
}
